sharktank/tests/models/llama4/moe_test.py

```diff
@@ -0,0 +1,220 @@
+from sharktank.utils.testing import TempDirTestBase
+from sharktank.layers.ffn_block import FFN
+import torch
+
+
+class Llama4Test(TempDirTestBase):
+    def setUp(self):
+        super().setUp()
+        torch.random.manual_seed(12345)
+
+    def test_moe(self):
+        from sharktank.layers.testing import make_random_moe_block_theta
+        from sharktank.layers import MoeBlock
+
+        dtype = torch.float32
+        feature_dim = 7
+        expert_hidden_dim = 3
+        num_experts = 5
+        expert_used_count = 2
+        num_shared_experts = 11
+        shared_expert_hidden_dim = 13
+        batch_size = 17
+        sequence_length = 19
+
+        theta = make_random_moe_block_theta(
+            block_idx=0,
+            in_dim=feature_dim,
+            expert_hidden_dim=expert_hidden_dim,
+            num_experts=num_experts,
+            with_ffn_norm=True,
+            num_shared_experts=num_shared_experts,
+            with_layer_output_norm=False,
+            dtype=dtype,
+        )
+
+        moe_block = MoeBlock(
+            theta=theta,
+            expert_count=num_experts,
+            expert_used_count=expert_used_count,
+            rms_epsilon=0.01,
+            moe_activation=torch.nn.functional.silu,
+            experts_ffn_moe_block="PreGatherFFNMOE",
+            score_experts=torch.nn.functional.sigmoid,
+            normalize_experts=False,
+            expert_shared_count=num_shared_experts,
+            model_arch="llama4",
+        )
+
+        input = (
+            torch.rand([batch_size, sequence_length, feature_dim], dtype=dtype) - 0.5
+        )
+        moe_output = moe_block(input)
+
+        from sharktank.types import unbox_tensor, Theta
+
+        if num_shared_experts is not None:
```

Comment:
Are you planning on paramaterizing this test to run with and without shared experts?

Reply:
For now, I have removed it.